shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java

```diff
@@ -225,17 +235,20 @@ public void ensureInitializationComplete(
         shellArgs.add("--log-tag=" + settings.getLogTag());
       }
 
+      ActivityManager activityManager =
+          (ActivityManager) applicationContext.getSystemService(Context.ACTIVITY_SERVICE);
+      int oldGenHeapSizeMegaBytes = activityManager.getLargeMemoryClass();
+
       long initTimeMillis = SystemClock.uptimeMillis() - initStartTimestampMillis;
 
-      if (FlutterInjector.instance().shouldLoadNative()) {
-        FlutterJNI.nativeInit(
-            applicationContext,
-            shellArgs.toArray(new String[0]),
-            kernelPath,
-            result.appStoragePath,
-            result.engineCachesPath,
-            initTimeMillis);
-      }
+      flutterJNILoader.nativeInit(
+          applicationContext,
+          shellArgs.toArray(new String[0]),
+          kernelPath,
+          result.appStoragePath,
+          result.engineCachesPath,
+          initTimeMillis,
+          oldGenHeapSizeMegaBytes);
```

Comment:
Can `oldGenHeapSizeMegaBytes` be provided as a shell command line flag in `shell/common/switches.h`?

Reply:
It could but then we lose some static type checking here. Is there some reason you have in mind to do it that way?

Reply:
It would be more consistent with how other engine settings are assigned and would avoid a proliferation of `nativeInit` arguments

Reply:
Done